docs/security/rbac-authorization.md

```diff
@@ -0,0 +1,240 @@
+    </ul>
+  </td>
+  <td>
+  </td>
+</tr>
+<tr>
+  <td>
+    <b>Contributor</b>
+  </td>
+  <td>
+    <ul>
+      <li>Can manually run, modify and stop workflows/jobs.</li>
+      <li>Can view project-level secrets and organization-wide secrets scoped for the given project.</li>
+      <li>Can attach to running jobs or debug jobs and projects.</li>
+      <li>Can view schedulers, project insights, and repository info.</li>
+      <li>Can manually run schedulers.</li>
+      <li>Can view, modify and delete artifacts for that project.</li>
+    </ul>
+  </td>
+  <td>
+  </td>
+</tr>
+<tr>
+  <td>
+    <b>Maintainer</b>
+  </td>
+  <td>
+    <ul>
+      <li>Can do everything a contributor can.</li>
+      <li>Can view and manage people within the project.</li>
+      <li>Can view modify and manage project-level secrets, schedulers and 
+      project level pre-flight checks.</li>
+      <li>Can view and manage project settings.</li>
+    </ul>
+  </td>
+  <td>
+  </td>
+</tr>
+<tr>
+  <td>
+    <b>Admin</b>
+  </td>
+  <td>
+    <ul>
+      <li>Can do everything within the project, including deleting it.</li>
+    </ul>
+  </td>
+  <td>
+    By default, this role is assigned to the user that created the project, and
+    this user is a primary repository token holder.
+  </td>
+</tr>
+</tbody>
+</table>
+
+## Role Management
+
+Users can get roles in a few different ways (from different sources):
+<ul>
+<li>Each user can get one organization role assigned to them directly (from the organization's Admin or Owner).</li>
+<li>User can get an organization-level role from membership within one (or more) organization groups.</li>
+</ul>
+
+Within each project user can be assigned a role:
+<ul>
+<li>Directly form projects Admin or Maintainer.</li>
+<li>From membership within one (or more) organization groups.</li>
+<li>From access rights within [project's repository](#repository-to-role-mappings) (GitHub or BitBucket).</li>
+<li>From the role that user has [within the organization](#organization-role-to-project-role-mappings).</li>
+</ul>
+
+TODO Picture from UI for assigning roles/viewing role assignments
+
+### Repository to role mappings
+
+If an organization member (someone who has any role within the organization) also has access to the projects
+remote repository on GitHub or BitBucket, that member will be assigned a role within the project automatically.
+
+<b>Pull</b> access within the repository -> Project <b>Reader</b> role
+<br/>
+<b>Push</b> access within the repository -> Project <b>Contributor</b> role
+<br/>
+<b>Admin</b> access within the repository -> Project <b>Maintainer</b> role
+<br/>
+
+### Organization role to project role mappings
+Having specific roles within the organization can grant you a role within all of the projects automatically.
+
+Organization <b>Owner</b> -> Project <b>Admin</b>
+<br/>
+Organization <b>Admin</b> -> Project <b>Admin</b>
+<br/>
+
+# Custom roles
```

Comment:
We also do not have this currently.

Reply:
Ok. I put this here in the draft for the purpose of conveying a full picture of what is RBAC intended to be.